centreon/src/Core/Common/Infrastructure/Upload/FileIterator.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types = 1);
+
+namespace Core\Common\Infrastructure\Upload;
+
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+class FileIterator
```

Comment:
Confusing naming problem here for me.

We have in the same folder
- `FileIteratorInterface` which is a php `Iterator`
- `FileIterator` which is a simple class which does not implement `FileIteratorInterface` and which is not an `Iterator`.

👉🏻 `FileList`, `FileCollection` ?